patch-package
--- a/node_modules/panini/lib/render.js
+++ b/node_modules/panini/lib/render.js
@@ -21,6 +21,10 @@ function render(file, enc, cb) {
     var page = fm(stripBom(file.contents.toString()));
     var pageData;
 
+    // Rendering is processed in asynchronous strem - therefore you need to refresh Handlebars Configuration when in multi-instance mode
+    this.Handlebars.partials = {};
+    this.refresh();
+
     // Determine which layout to use
     var basePath = path.relative(this.options.root, path.dirname(file.path));
     var layout =
--- a/node_modules/panini/lib/utils.js
+++ b/node_modules/panini/lib/utils.js
@@ -16,7 +16,12 @@ exports.loadFiles = function(dir, pattern) {
   dir = !Array.isArray(dir) ? [dir] : dir;
 
   for (var i in dir) {
-    files = files.concat(glob.sync(path.join(process.cwd(), dir[i], pattern)));
+    var filePath = [process.cwd(), dir[i], pattern];
+    if(path.isAbsolute(dir[i])) {
+      filePath.shift();
+    }
+
+    files = files.concat(glob.sync(path.join.apply(null, filePath)));
   }
 
   return files;
